// ngx-isotope/__ivy_ngcc__/ngx-isotope.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/core'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['@angular/core','@angular/common','exports', '@angular/core', '@angular/common'], factory) :
	(factory(global.ng.core,global.ng.common,(global['ngx-isotope'] = {}),global._angular_core,global._angular_common));
}(this, (function (ɵngcc0,ɵngcc1,exports,_angular_core,_angular_common) { 
var _c0 = ["*"];
'use strict';

var IsotopeGridComponent = (function () {
    /**
     * @param {?} el
     */
    function IsotopeGridComponent(el) {
        this.el = el;
        this.items = [];
    }
    /**
     * @return {?}
     */
    IsotopeGridComponent.prototype.ngOnInit = function () {
        if (!this.options)
            this.options = {};
        if (!this.options.itemSelector) {
            this.options.itemSelector = '[isotope-item], isotope-item';
        }
        if (this.el.nativeElement.tagName === 'ISOTOPE-GRID') {
            this.el.nativeElement.style.display = 'block';
        }
        this.isotope = new Isotope(this.el.nativeElement, this.options);
    };
    /**
     * @param {?} el
     * @return {?}
     */
    IsotopeGridComponent.prototype.add = function (el) {
        var _this = this;
        this.isotope.appended(el);
        this.isotope.layout();
        imagesLoaded(el).on('progress', function () {
            _this.isotope.layout();
        });
    };
IsotopeGridComponent.ɵfac = function IsotopeGridComponent_Factory(t) { return new (t || IsotopeGridComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
IsotopeGridComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IsotopeGridComponent, selectors: [["isotope-grid"]], inputs: { options: "options" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function IsotopeGridComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsotopeGridComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'isotope-grid',
                template: "<ng-content></ng-content> ",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { options: [{
            type: _angular_core.Input
        }] }); })();
    return IsotopeGridComponent;
}());
/**
 * @nocollapse
 */
IsotopeGridComponent.ctorParameters = function () { return [
    { type: _angular_core.ElementRef, },
]; };
IsotopeGridComponent.propDecorators = {
    'options': [{ type: _angular_core.Input },],
};

var IsotopeItemComponent = (function () {
    /**
     * @param {?} grid
     * @param {?} el
     */
    function IsotopeItemComponent(grid, el) {
        this.grid = grid;
        this.el = el;
    }
    /**
     * @return {?}
     */
    IsotopeItemComponent.prototype.ngOnInit = function () {
    };
    /**
     * @return {?}
     */
    IsotopeItemComponent.prototype.ngAfterViewInit = function () {
        this.grid.add(this.el.nativeElement);
    };
IsotopeItemComponent.ɵfac = function IsotopeItemComponent_Factory(t) { return new (t || IsotopeItemComponent)(ɵngcc0.ɵɵdirectiveInject(IsotopeGridComponent), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
IsotopeItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IsotopeItemComponent, selectors: [["isotope-item"]], ngContentSelectors: _c0, decls: 1, vars: 0, template: function IsotopeItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsotopeItemComponent, [{
        type: _angular_core.Component,
        args: [{
                selector: 'isotope-item',
                template: "<ng-content></ng-content> ",
                styles: [""]
            }]
    }], function () { return [{ type: IsotopeGridComponent }, { type: ɵngcc0.ElementRef }]; }, null); })();
    return IsotopeItemComponent;
}());
/**
 * @nocollapse
 */
IsotopeItemComponent.ctorParameters = function () { return [
    { type: IsotopeGridComponent, },
    { type: _angular_core.ElementRef, },
]; };

var IsotopeModule = (function () {
    function IsotopeModule() {
    }
    /**
     * @return {?}
     */
    IsotopeModule.forRoot = function () {
        return {
            ngModule: IsotopeModule,
            providers: []
        };
    };
IsotopeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IsotopeModule });
IsotopeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IsotopeModule_Factory(t) { return new (t || IsotopeModule)(); }, imports: [[
            _angular_common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IsotopeModule, { declarations: [IsotopeGridComponent, IsotopeItemComponent], imports: [ɵngcc1.CommonModule], exports: [IsotopeGridComponent, IsotopeItemComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsotopeModule, [{
        type: _angular_core.NgModule,
        args: [{
                imports: [
                    _angular_common.CommonModule
                ],
                declarations: [
                    IsotopeGridComponent,
                    IsotopeItemComponent,
                ],
                exports: [
                    IsotopeGridComponent,
                    IsotopeItemComponent,
                ]
            }]
    }], function () { return []; }, null); })();
    return IsotopeModule;
}());
/**
 * @nocollapse
 */
IsotopeModule.ctorParameters = function () { return []; };

exports.IsotopeModule = IsotopeModule;
exports.IsotopeGridComponent = IsotopeGridComponent;
exports.IsotopeItemComponent = IsotopeItemComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
